SSA/Projects/InstCombine/ForMathlib.lean

```diff
@@ -472,6 +469,11 @@ inductive Refinement {α : Type u} : Option α → Option α → Prop
   | bothSome {x y : α } : x = y → Refinement (some x) (some y)
   | noneAny {x? : Option α} : Refinement none x?
 
+@[simp]
```

Comment:
I would not tag this as `simp` here, because it breaks the abstraction of our simp_mlir. I would add this to the `simp` set above in the tactic.

Reply:
Sure, feel free to edit this.